### PR TITLE
Relationship artifacts (2)

### LINKED ISSUE #175: Relationship artifacts (2)
The property "full_product_name" in items of the array "relationships" should be required as stated in the specification of CVRF 1.2 (see CSAF-5.1.3-1). Otherwise, we don't have a name to reference the relationship. #89 didn't address that.

### PR DESCRIPTION
**Pull request by tschmidtb51**
- resolves #175 
- add "full_product_name" to required fields in items of "relationships" to be compatible to CVRF 1.2 (see CSAF-5.1.3-1)
- addresses prose related to #175, #99
- change prose to reflect schema

Note: This does not change the behavior introduced by #99.

### CODE REVIEW

**Review by sthagen — APPROVED** [review]
LGTM

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 1]
```diff
@@ -1302,9 +1302,9 @@ List of relationships (`relationships`) of value type `array`with 1 or more item
       }
     }
 
-The Relationship item is of value type `object` with the three mandatory properties Product Reference (`product_reference`), Relates to Product Reference (`relates_to_product_reference`), and Relationship Type (`relationship_type`) as well as the optional property Full Product Names (`full_product_names`) establishes a link between two existing `full_product_name_t` elements, allowing the document producer to define a combination of two products that form a new `full_product_name` entry.
+The Relationship item is of value type `object` with the four mandatory properties Full Product Name (`full_product_name`), Product Reference (`product_reference`), Relates to Product Reference (`relates_to_product_reference`), and Relationship Type (`relationship_type`) establishes a link between two existing `full_product_name_t` elements, allowing the document producer to define a combination of two products that form a new `full_product_name` entry.
```
Comment (mprpic) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1305]: I'm having a hard time parsing this sentence. Should there be a colon after "four mandatory properties" and the "establishes a link..." part maybe needs a separate sentence?

How about:

```suggestion
The Relationship item is of value type `object` with four mandatory properties: Full Product Name (`full_product_name`), Product Reference (`product_reference`), Relates to Product Reference (`relates_to_product_reference`), and Relationship Type (`relationship_type`). The Relationship item establishes a link between two existing `full_product_name_t` elements, allowing the document producer to define a combination of two products that form a new `full_product_name` entry.
```

If it reads fine for you, feel free to ignore this suggestion ;-)

**Review by mprpic — APPROVED** [review]

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 1 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md]: Thanks for the suggestion. I just pushed an additional commit.

Comment (mprpic) [reply csaf_2.0/prose/csaf-v2-editor-draft.md]: Excellent, thank you!

**Review by sthagen — APPROVED** [review]

**Review by santosomar — APPROVED** [review]
Approved with no concerns.

**Pull request merged**